CMakeLists.txt

```diff
@@ -34,11 +34,20 @@ list ( APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/" )
 # Find Julia
 find_package( Julia REQUIRED )
 
-
-
 # Find MPI
 find_package( MPI REQUIRED )
 
+# Find google test on demand
+option( TESTING "Build tests using google test" )
+if( TESTING )
+    find_package( GTest REQUIRED )
+    if ( NOT DEFINED JULIA_PROJECT_PATH )
+        message( WARNING "JULIA_PROJECT_PATH not set, tests will not work.")
```

Comment:
Why not make this an error right now? Is there any conceivable use case where you'd want to enable tests but be OK with them failing?

Reply:
Actually not, I think.